### PR TITLE
Use proxy config for Google client configuration

### LINKED ISSUE #68: OAuth doesn't work behind a proxy / Plugin doesn't use proxy configuration
When behind a proxy, OAuth doesn't work, because the request back to google fails. This is because, as far as I understood, uses its own implementation of making this request and therefore doesn't use the configured proxy.

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

fixes #68 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by rolfcronberg** [comment]
It seem to work. The plugin use the proxy but I get another error message. 
--------------------------------------------------------
ERROR [2021-09-16 12:15:30] 21946  Uncaught exception: 
/usr/share/matomo/plugins/GoogleAnalyticsImporter/vendor/google/apiclient/src/Http/REST.php(128): 
{
   "error": {
     "code": 403,
     "message": "User does not have permission to perform this operation.",
     "errors": [
       {
         "message": "User does not have permission to perform this operation.",
         "domain": "global",
         "reason": "insufficientPermissions"
       }
     ]
   }
}
  [Query: , CLI mode: 1]

**Comment by sgiehl** [comment]
That might be unrelated to the proxy. I'm not really deep in the Google Analytics import. maybe @diosmosis has a quick idea on that?

**Comment by diosmosis** [comment]
Sounds like the GA user you are logging in with doesn't have access to the GA property.

**Comment by rolfcronberg** [comment]
I must have missed something, I am an administrator for the property.

File: config/config.php [hunk 1]
```diff
@@ -30,6 +31,26 @@
...
+            $proxyPort     = Config::getInstance()->proxy['port'];
+            $proxyUser     = Config::getInstance()->proxy['username'];
+            $proxyPassword = Config::getInstance()->proxy['password'];
+
+            if ($proxyUser) {
+                $proxy = sprintf('http://%s:%s@%s:%s', $proxyUser, $proxyPassword, $proxyHost, $proxyPort);
+            } else {
+                $proxy = sprintf('http://%s:%s', $proxyHost, $proxyPort);
+            }
+            $httpClient = new \GuzzleHttp\Client([
```
Comment (bhayden-umn) [inline config/config.php]: Changed this to `new GuzzleHttp\Client` to work in Matomo 5.0.3/PHP 8.2.15

File: config/config.php [hunk 2]
```diff
@@ -30,6 +31,26 @@
...
+                'proxy'      => $proxy,
+                'exceptions' => false,
+                'base_uri'   => \Google_Client::API_BASE_PATH
```
Comment (bhayden-umn) [inline config/config.php]: Changed this to `Google_Client` to work in Matomo 5.0.3/PHP 8.2.15

**Review by bhayden-umn — CHANGES_REQUESTED** [review]

**Comment by bhayden-umn** [comment]
@AltamashShaikh Let me know if I can help to move this forward. Tagging you since it seems you're the most active maintainer recently, not sure if @sgiehl is still contributing?

**Comment by snake14** [comment]
Hi @machinehum I rebased the PR to 5.x-dev. Can you please confirm that it works for you with the latest changes to this PR? It works fine when I test it, but I don't have a proxy set up. If you can confirm that it works with a proxy, I can merge this PR.

**Comment by bhayden-umn** [comment]
@snake14 Applied this and it works as expected!

**Comment by bhayden-umn** [comment]
@snake14 I've seen commentary in a few places about folks not having a proxy setup easily available for testing. Would it be helpful to contribute something like a docker-compose setup that runs the official Matomo docker image in one container, through a proxy in another container, for testing purposes?

**Comment by snake14** [comment]
Hi @machinehum . Thank you for testing that and confirming that it works with a proxy.

Yes. If you have time to configure a docker setup for testing a proxy, that would be greatly appreciated. I think that the one requirement would be that the file system be accessible or the plugin branch configurable so that we could test specific PRs before they're merged.

**Review by snake14 — APPROVED** [review]
This code is actually very similar to what we're already using in part of the SearchEngineKeywordsPerformance plugin. The specific PR has been tested and confirmed to work.

**Pull request merged**